### ISSUE TITLE
[Bug] Upgrade nC 13.0.2 to 14.0.4 Could not write to .step

### ISSUE THREAD

**Issue by BloodyIron** [nextcloud/server#12770]
1. Ubuntu 18.04 Server
2. Apache2 2.4.29
3. mysql Ver 14.14 Distrib 5.7.24, for Linux (x86_64) using  EditLine wrapper
4. PHP 7.2
5. nextCloud 13.0.2 (current) 14.0.4 (target)

My data folder is an SMB mount, with www-data:www-data permissions set recursively in fstab. nextCloud has zero issues making files/folders for users in here, and the /update-instanceid/ folder has the same permissions as the other sections. Permissions are 770 btw.

I've rebooted the server, restarted apache, remounted the SMB share, and I don't know why it can't make the .snap file in the /update-instanceid/ folder in the data mount.

Reproduce:
1. Go to admin section of my nextCloud instance
2. Go to "Basic Settings"
3. Click on "Open Updater"
4. Click "Start Update"
5. Error produced "Check for expected files Could not write to .step"

I would expect it to have permissions to write these files, because at the filesystem level it does and can write files and folders in other parts of the data mount.

Seriously out of ideas here :(

**Comment by BloodyIron** [comment]
Can't upgrade my nextCloud till I solve this :( Anyone with ideas?

**Comment by BloodyIron** [comment]
Anyone? I can't upgrade...

**Comment by BloodyIron** [comment]
Okay, so I found my "solution".

The documents (data) folder is actually an SMB/CIFS share mounted onto a location within the Linux VM that nextCloud runs in. The SMB/CIFS share is backed by FreeNAS, and the share has a setting "Show Hidden Files" that was unchecked (this is a FreeNAS setting for the SMB/CIFS share for my data/documents content).

I checked the "Show Hidden Files" setting ON, hit "OK" (FreeNAS here), and that cycled Samba on the FreeNAS end. Then I stopped apache2 on the nextCloud server, unmounted the data folder, remounted it, and started apache2. I then went to the webGUI to trigger and upgrade, and BAM now it is past the ".step" part.

I'm now at a gateway timeout issue when the backup is made, so I'm further along. Can probably solve this other issue...

Looks like I'm literally the only person on the internet with this issue, so hope this helps someone!
